frontend/src/lib/uPlotLib/getUplotChartData.ts

```diff
@@ -118,33 +118,31 @@ const createDivsFromArray = (
 			} else {
 				const color = colors[(index - 1) % colors.length];
 
-				if (data[idx]) {
-					const squareBox = document.createElement('div');
-					squareBox.classList.add('pointSquare');
+				const squareBox = document.createElement('div');
```

Comment:
We should have this check to avoid creating unwanted div s when data isn't present

Reply:
this is already controlled via

```tsx
if (item.show) {
					div.appendChild(squareBox);
					div.appendChild(text);
				}
```